src/openvic-simulation/history/CountryHistory.cpp

```diff
@@ -0,0 +1,308 @@
+	return government_type;
+}
+
+const fixed_point_t CountryHistory::get_plurality() const {
+	return plurality;
+}
+
+NationalValue const* CountryHistory::get_national_value() const {
+	return national_value;
+}
+
+const bool CountryHistory::is_civilised() const {
+	return civilised;
+}
+
+const fixed_point_t CountryHistory::get_prestige() const {
+	return prestige;
+}
+
+const std::vector<Reform const*>& CountryHistory::get_reforms() const {
+	return reforms;
+}
+
+OOB const* CountryHistory::get_inital_oob() const {
+	return inital_oob;
+}
+
+#define EXISTING_ENTRY country_histories.at(country).at(date)
+
+bool CountryHistoryManager::add_country_history_entry(
+	Country const* country,
+	const Date date,
+	Culture const* primary_culture,
+	const std::vector<Culture const*>&& accepted_cultures,
+	Religion const* religion,
+	CountryParty const* ruling_party,
+	const Date last_election,
+	const std::map<Ideology const*, const fixed_point_t>&& upper_house,
+	Province const* capital,
+	GovernmentType const* government_type,
+	const fixed_point_t plurality,
+	NationalValue const* national_value,
+	const bool civilised,
+	const fixed_point_t prestige,
+	const std::vector<Reform const*>&& reforms,
+	OOB const* initial_oob,
+	bool updated_accepted_cultures,
+	bool updated_upper_house,
+	bool updated_reforms
+) {
+	if (locked) {
+		Logger::error("Cannot add new history entry to country history registry: locked!");
+		return false;
+	}
+
+	/* create country entry if doesn't already exist */
+	if (country_histories.find(country) == country_histories.end()) country_histories[country];
+
+	/* combine duplicate histories, priority to current (defined later) */
+	if (country_histories.at(country).find(date) != country_histories.at(country).end()) {
+		CountryHistory entry = CountryHistory {
+			primary_culture == nullptr ? 	EXISTING_ENTRY.primary_culture : 	primary_culture,
+			!updated_accepted_cultures ? 	EXISTING_ENTRY.accepted_cultures : 	std::move(accepted_cultures),
+			religion == nullptr ? 			EXISTING_ENTRY.religion : 			religion,
+			ruling_party == nullptr ? 		EXISTING_ENTRY.ruling_party : 		ruling_party,
+			last_election == Date(0) ? 		EXISTING_ENTRY.last_election : 		last_election,
+			!updated_upper_house ? 			EXISTING_ENTRY.upper_house : 		std::move(upper_house),
+			capital == nullptr ? 			EXISTING_ENTRY.capital : 			capital,
+			government_type == nullptr ?	EXISTING_ENTRY.government_type : 	government_type,
+			plurality == -1 ?				EXISTING_ENTRY.plurality :			plurality,
+			national_value == nullptr ?		EXISTING_ENTRY.national_value :		national_value,
+			civilised == false ?			EXISTING_ENTRY.civilised :			civilised,
+			prestige == -1 ?				EXISTING_ENTRY.prestige :			prestige,
+			!updated_reforms ?				EXISTING_ENTRY.reforms :			std::move(reforms),
+			initial_oob == nullptr ?		EXISTING_ENTRY.inital_oob :			initial_oob
+		};
+
+		country_histories.at(country).erase(date);
+		country_histories.at(country).emplace(date, entry);
```

Comment:
Why not: (I'm using `auto` to shorthand this, use the actual iterator types, even if you must use `decltype`)
```suggestion
	auto country_history = country_histories[country];
	auto date_history = country_history.find(date);

	/* combine duplicate histories, priority to current (defined later) */
	if (date_history != country_history.end()) {
		CountryHistory entry = CountryHistory {
			primary_culture == nullptr ? 	date_history->primary_culture : 	primary_culture,
			!updated_accepted_cultures ? 	date_history->accepted_cultures : 	std::move(accepted_cultures),
			religion == nullptr ? 			date_history->religion : 			religion,
			ruling_party == nullptr ? 		date_history->ruling_party : 		ruling_party,
			last_election == Date(0) ? 		date_history->last_election : 		last_election,
			!updated_upper_house ? 			date_history->upper_house : 		std::move(upper_house),
			capital == nullptr ? 			date_history->capital : 			capital,
			government_type == nullptr ?	date_history->government_type : 	government_type,
			plurality == -1 ?				date_history->plurality :			plurality,
			national_value == nullptr ?		date_history->national_value :		national_value,
			civilised == false ?			date_history->civilised :			civilised,
			prestige == -1 ?				date_history->prestige :			prestige,
			!updated_reforms ?				date_history->reforms :			std::move(reforms),
			initial_oob == nullptr ?		date_history->inital_oob :			initial_oob
		};

		country_history.erase(date);
		country_history.emplace(date, entry);
```

Also this is a really valid case why const'ing everything isn't always a great idea, if we're gonna be modifying an existing object, all this does is create useless construction, in which case it be more efficient not to const these values, which given we already make them private variables anyway never really made sense to me.